test/templates/templates_register_api_test.rb

```diff
@@ -0,0 +1,28 @@
+require 'test_helper'
+require 'json'
+require 'templates/templates_register_api'
+require 'templates/templates'
+require 'webmock/test_unit'
+
+class TemplatesRegisterApiTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    Proxy::TemplatesRegisterApi.new
+  end
+
+  def setup
+    @foreman_url = 'https://foreman.example.com'
+    Proxy::SETTINGS.stubs(:foreman_url).returns(@foreman_url)
+    @template_url = 'https://foreman2.example.com'
```

Comment:
Perhaps this example is more realistic. Doesn't really matter for the actual result, but in the future it'll be easier to understand what's going on.
```suggestion
    @template_url = 'http://smart-proxy.example.com:8000'
```